binary_transparency/firmware/cmd/ft_monitor/impl/ft_monitor.go

```diff
@@ -62,132 +59,71 @@ func Main(ctx context.Context, opts MonitorOpts) error {
 		return fmt.Errorf("failed to parse FT log URL: %w", err)
 	}
 
-	glog.Infof("Monitoring FT log %q...", opts.LogURL)
-	ticker := time.NewTicker(opts.PollInterval)
-
-	c := client.ReadonlyClient{LogURL: ftURL}
-	var latestCP api.LogCheckpoint
-
-	lv := verify.NewLogVerifier()
-
 	// Parse the input keywords as regular expression
 	var ftKeywords []*regexp.Regexp
 	for _, k := range strings.Fields(opts.Keyword) {
 		ftKeywords = append(ftKeywords, regexp.MustCompile(k))
 	}
 
-	for {
+	c := client.ReadonlyClient{LogURL: ftURL}
+
+	// TODO(mhutchinson): This checkpoint and tracker for number of processed entries
+	// should be serialized so the monitor persists its golden state between runs.
+	var latestCP api.LogCheckpoint
+	var head uint64
+	follow := client.NewLogFollower(c)
+
+	glog.Infof("Monitoring FT log %q...", opts.LogURL)
+	cpc, cperrc := follow.Checkpoints(ctx, opts.PollInterval, latestCP)
+	ec, eerrc := follow.Entries(ctx, cpc, head)
+
+	for entry := range ec {
 		select {
-		case <-ticker.C:
-			//
+		case err = <-cperrc:
+			return err
+		case err = <-eerrc:
+			return err
 		case <-ctx.Done():
 			return ctx.Err()
+		default:
 		}
```

Comment:
I'm a bit squinty about this - I think the logic of channels/errs and select works, but it's potentially a bit brittle e.g. if someone later decides that sending an informative error down `cperrc` without bailing out of the goroutine and closing the `cpc` channel is ok.

Wondering if it might be better/more readable to do something like this instead? :
```golang

cpc, cperrc := follow.Checkpoints(ctx, opts.PollInterval, latestCP)
ec, eerrc := follow.Entries(ctx, cpc, head)

for {
  select {
     // handle errs
     case entry, ok := <- ec:
       if !ok {
          break
   }
}
```

Reply:
I don't have the `ok` return, but I think the new version is more consistent and clearer. PTAL.